examples/cc3200/xively_demo/main.c

```diff
@@ -0,0 +1,1044 @@
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Green LED";
+const char* const gOrangeLedTopicName =
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Orange LED";
+const char* const gRedLedTopicName =
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Red LED";
+const char* const gButtonSW2TopicName =
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Button SW2";
+const char* const gButtonSW3TopicName =
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Button SW3";
+
+/* Required by push button interrupt handlers - there is no space for user argument so we
+ * can't pass the xively's context handler as a function argument. */
+xi_context_handle_t gXivelyContextHandle = -1;
+
+/* Used to store the temperature timed task handles. */
+xi_timed_task_handle_t gSendTempratureHandle = -1;
+
+/* Used by onLed handler function to differentiate LED's */
+const static ledNames gGreenLed  = MCU_GREEN_LED_GPIO;
+const static ledNames gOrangeLed = MCU_ORANGE_LED_GPIO;
+const static ledNames gRedLed    = MCU_RED_LED_GPIO;
+
+/******************************************************************************/
+/* END OF XIVELY CODE GLOBAL SECTION */
+/******************************************************************************/
+
+/**
+* @brief Event handler for SW2 button press/release interrupts.
+ *
+ * If the button was pushed the pinValue will be 1 if it was released 0.
+ * @note This function uses the global variable of gXivelyContextHandle - the button_if
+ * implementation doesn't provide alternative.
+ */
+void pushButtonInterruptHandlerSW2()
+{
+    int pinValue = GPIOPinRead( GPIOA2_BASE, GPIO_PIN_6 );
+
+    if ( pinValue == 0 )
+    {
+        xi_publish( gXivelyContextHandle, gButtonSW2TopicName, "0",
+                    XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
+    }
+    else
+    {
+        xi_publish( gXivelyContextHandle, gButtonSW2TopicName, "1",
+                    XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
+    }
+
+    Button_IF_EnableInterrupt( SW2 );
+}
+
+/**
+ * @brief Event handler for SW2 button press/release interrupts.
+ *
+ * If the button was pushed the pinValue will be 1 if it was released 0.
+ * @note This function uses the global variable of gXivelyContextHandle - the button_if
+ * implementation doesn't provide alternative.
+ */
+void pushButtonInterruptHandlerSW3()
+{
+    int pinValue = GPIOPinRead( GPIOA1_BASE, GPIO_PIN_5 );
+
+    if ( pinValue == 0 )
+    {
+        xi_publish( gXivelyContextHandle, gButtonSW3TopicName, "0",
+                    XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
+    }
+    else
+    {
+        xi_publish( gXivelyContextHandle, gButtonSW3TopicName, "1",
+                    XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
+    }
+
+    Button_IF_EnableInterrupt( SW3 );
+}
+
+/**
+ * @brief Function that handles events related to the LEDs' topics messages and

```

Comment:
Function that handles messages sent to the device on the LEDs' topics, including toggle commands and subscription changes.